### PR TITLE
Add more tests and allow partly run

### PR DESCRIPTION
**Pull request by viktorprogger**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #27

### CODE REVIEW

File: tests/unit/CallableJobTest.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Tests\unit;
+
+use stdClass;
+use Yiisoft\Serializer\SerializerInterface;
+use Yiisoft\Yii\Queue\Job\CallableJob;
+use Yiisoft\Yii\Queue\Tests\TestCase;
+
+class CallableJobTest extends TestCase
```
Comment (xepozz) [inline tests/unit/CallableJobTest.php:12]: How about to use DataProvider?
Also would be nice to test non-static closures

File: src/Queue.php [hunk 2]
```diff
@@ -103,13 +103,17 @@ public function push(JobInterface $job): string
 
     /**
      * Execute all existing jobs and exit
+     *
+     * @param int $max
      */
-    public function run(): void
+    public function run(int $max = 0): void
```
Comment (xepozz) [inline src/Queue.php:109]: Would be nice to check that `$max` greater than zero or equals to zero

File: tests/unit/ExceptionTest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+use Yiisoft\Yii\Queue\Driver\SynchronousDriver;
+use Yiisoft\Yii\Queue\Exception\JobNotSupportedException;
+use Yiisoft\Yii\Queue\Job\DelayableJobInterface;
+use Yiisoft\Yii\Queue\Job\PrioritisedJobInterface;
+use Yiisoft\Yii\Queue\Job\RetryableJobInterface;
+use Yiisoft\Yii\Queue\Tests\App\DelayableJob;
+use Yiisoft\Yii\Queue\Tests\TestCase;
+
+class ExceptionTest extends TestCase
```
Comment (xepozz) [inline tests/unit/ExceptionTest.php:15]: Can it be marked as `final`?

**Comment by xepozz** [comment]
> Fixed issues	#46

Incorrect issue specified

**Comment by viktorprogger** [comment]
@xepozz thank you for your comments

**Comment by samdark** [comment]
Overall looks fine so far.

File: tests/unit/ExceptionTest.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Tests\unit;
```
Comment (romkatsu) [inline tests/unit/ExceptionTest.php:5]: Better fix namespace Yiisoft\Yii\Queue\Tests\Unit;

File: src/Queue.php (hunk 2 above)
Comment (viktorprogger) [reply src/Queue.php:115]: In the current implementation it doesn't matter if `$max` is zero or less then zero.

**Comment by roxblnfk** [comment]
`infection.log` should be ignored

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**